lib/taurus/qt/qtgui/taurusgui/utils.py

```diff
@@ -365,7 +365,23 @@ class PanelDescription(TaurusGuiComponentDescription):
     '''
 
     def __init__(self, *args, **kwargs):
+        """
+
+        :param args:
+        :param instrumentkey: (str)
+        :param modelinconfig: (bool) whther to store model in settigns file or not
+        :param modifiablebyuser: (bool) whether user can edit widget or not
+        :param formatter: (str) formatter used by this widget
+
+        Additionally, extra configuration options can be passed in constructor as key word arguments.
+        Proper widget attributes will be set to corresponding values
+        """
         self.instrumentkey = kwargs.pop('instrumentkey', None)
+        self.icon = kwargs.pop("icon", None)
+        self.model_in_config = kwargs.pop("modelinconfig", False)
+        self.modifiable_by_user = kwargs.pop("modifiablebyuser", False)
+        self.formatter = kwargs.pop("formatter", 'taurus.qt.qtgui.base.taurusbase.defaultFormatter')
+        self.extra_config = kwargs
```

Comment:
In this proposed implementation, `extra_config` (which refers to widget attributes/properties to be set) is obtained implicitly as whatever kwargs not explicitly consumed in the `__init__`. In my experience it is better to require the caller to be explicit (e.g., by calling the constructor as `PanelDescriptor(icon=..., widget_properties={"foo"=1,"bar":2})`.

While the implicit version is more concise, it also becomes a PITA to maintain (think of refactorings, deprecations, etc.)

Reply:
I trust your experience. Will change.

I think `widget_properties` is better (more descriptive) name for these options